src/test/statistics/chunk_statistics/counting_quotient_filter_test.cpp

```diff
@@ -0,0 +1,134 @@
+    int_cqf16 = std::make_shared<CountingQuotientFilter<int>>(4, 16);
+    int_cqf32 = std::make_shared<CountingQuotientFilter<int>>(4, 32);
+    int_cqf2->populate(int_column);
+    int_cqf4->populate(int_column);
+    int_cqf8->populate(int_column);
+    int_cqf16->populate(int_column);
+    int_cqf32->populate(int_column);
+  }
+
+  std::shared_ptr<CountingQuotientFilter<std::string>> string_cqf2;
+  std::shared_ptr<CountingQuotientFilter<std::string>> string_cqf4;
+  std::shared_ptr<CountingQuotientFilter<std::string>> string_cqf8;
+  std::shared_ptr<CountingQuotientFilter<std::string>> string_cqf16;
+  std::shared_ptr<CountingQuotientFilter<std::string>> string_cqf32;
+  std::shared_ptr<CountingQuotientFilter<int>> int_cqf2;
+  std::shared_ptr<CountingQuotientFilter<int>> int_cqf4;
+  std::shared_ptr<CountingQuotientFilter<int>> int_cqf8;
+  std::shared_ptr<CountingQuotientFilter<int>> int_cqf16;
+  std::shared_ptr<CountingQuotientFilter<int>> int_cqf32;
+  std::shared_ptr<ValueColumn<std::string>> string_column;
+  std::shared_ptr<ValueColumn<int>> int_column;
+  std::map<std::string, size_t> string_value_counts;
+  std::map<int, size_t> int_value_counts;
+
+  template <typename DataType>
+  void test_value_counts(std::shared_ptr<CountingQuotientFilter<DataType>> cqf, std::map<DataType, size_t>
+      value_counts) {
+    for (auto value_and_count : value_counts) {
+        EXPECT_TRUE(cqf->count(value_and_count.first) >= value_and_count.second);
+    }
+  }
+
+  template <typename DataType>
+  void test_can_prune(std::shared_ptr<CountingQuotientFilter<DataType>> cqf, std::map<DataType, size_t>
+      value_counts) {
+    for (auto value_and_count : value_counts) {
+        EXPECT_FALSE(cqf->can_prune(value_and_count.first, PredicateCondition::Equals));
+    }
+  }
+};
+
+TEST_F(CountingQuotientFilterTest, NoUndercountsString) {
+  test_value_counts<std::string>(string_cqf2, string_value_counts);
+  test_value_counts<std::string>(string_cqf4, string_value_counts);
+  test_value_counts<std::string>(string_cqf8, string_value_counts);
+  test_value_counts<std::string>(string_cqf16, string_value_counts);
+  test_value_counts<std::string>(string_cqf32, string_value_counts);
+}
+
+TEST_F(CountingQuotientFilterTest, NoUndercountsInt) {
+  test_value_counts<int>(int_cqf2, int_value_counts);
+  test_value_counts<int>(int_cqf4, int_value_counts);
+  test_value_counts<int>(int_cqf8, int_value_counts);
+  test_value_counts<int>(int_cqf16, int_value_counts);
+  test_value_counts<int>(int_cqf32, int_value_counts);
+}
+
+TEST_F(CountingQuotientFilterTest, CanPruneString) {
+  test_can_prune<std::string>(string_cqf2, string_value_counts);
+  test_can_prune<std::string>(string_cqf4, string_value_counts);
+  test_can_prune<std::string>(string_cqf8, string_value_counts);
+  test_can_prune<std::string>(string_cqf16, string_value_counts);
+  test_can_prune<std::string>(string_cqf32, string_value_counts);
+}
+
+TEST_F(CountingQuotientFilterTest, CanPruneInt) {
```

Comment:
Any reason why you don't use GoogleTest's typed tests?

Would like to ensure that we cover all supported data types.

Reply:
I can look into that.

Reply:
I didn't see the how typed tests would help us here.

Reply:
You could automate (at least) the testing of all numerical values. Right now, we only test `int` which is insufficient when using such kind of "black box libraries" IMO.